### ISSUE TITLE
Solaris / SmartOS - Server console goes unresponsive after running command

### ISSUE THREAD

**Issue by brianewell** [PaperMC/Paper#1690]
### What behaviour is expected:
```
>say Can you hear me?
[22:12:59 INFO]: [Server] Can you hear me?
>
```
Command is entered and executed command prompt renders.  Further commands show up when entered.  Logging data from game continues uninterrupted.  This behavior occurs on vanilla, and on Spigot.

### What behaviour is observed:
```
> say Can you hear me?
[22:11:47 ERROR]: Caught previously unhandled exception :
[22:11:47 ERROR]: java.io.IOError: java.io.IOException: Error executing 'stty icrnl ixon echo icanon eof ^D eol ^A min 11111111111111111111111111111111 time 11111111111111111111111111111111': stty: invalid integer argument: '11111111111111111111111111111111': Value too large for defined data type

```
Command is entered but fails to execute.  Exception is raised.  Command prompt is not rendered again.  Further commands fail to show when typing.  Logging data from game continues uninterrupted.  This behavior occurs on paper.

### Steps/models to reproduce:
I'm running the following java options on a test server:
```java -Xms4G -Xmx4G -XX:+UseG1GC -XX:+UnlockExperimentalVMOptions -XX:MaxGCPauseMillis=100 -XX:+DisableExplicitGC -XX:TargetSurvivorRatio=90 -XX:G1NewSizePercent=50 -XX:G1MaxNewSizePercent=80 -XX:G1MixedGCLiveThresholdPercent=35 -XX:+AlwaysPreTouch -XX:+ParallelRefProcEnabled -Dusing.aikars.flags=mcflags.emc.gs -jar server.jar```

### Plugin list:
No plugins are installed on this test server.

### Paper build number:
```
This server is running Paper version git-Paper-451 (MC: 1.13.2) (Implementing API version 1.13.2-R0.1-SNAPSHOT)
Previous version: git-Paper-448 (MC: 1.13.2)
Checking version, please wait...
You are running the latest version
```

### Anything else:
The operating environment is Illumos/SmartOS:
```
System Info: Java 1.8 (OpenJDK 64-Bit Server VM 25.181-b00) Host:  SunOS 5.11 (amd64)
```

Attempting to execute the stty command directly on the shell yielded the same results:
```
$ stty icrnl ixon echo icanon eof ^D eol ^A min 11111111111111111111111111111111 time 11111111111111111111111111111111
stty: invalid integer argument: '11111111111111111111111111111111': Value too large for defined data type
```
Why does paper need 11 nonillion character minimum for a completed read, or a 3.52x10^22 year timeout on input?  These values seem kind of off.

**Comment by zachbr** [comment]
Thanks for reporting, we do not regularly test against Solaris/sunOS and derivatives, hopefully we can find the issue and solve it.

**Comment by electronicboy** [comment]
In the meantime, you could try to use the -nojline argument to disable the fancy terminal, I'd imagine the fallback would work fine here

**Comment by brianewell** [comment]
@zachbr It's Solaris/SunOS with a GNU userland, my stty is specifically from GNU coreutils 8.29.

@electronicboy thanks.  It's not quite as pretty, but it works.

```
say Can you hear me?
[22:49:21 INFO]: [Server] Can you hear me?

```

For anyone else's reference, my command parameters are as follows:
```
java -Xms6G -Xmx6G -XX:+UseG1GC -XX:+UnlockExperimentalVMOptions -XX:MaxGCPauseMillis=100 -XX:+DisableExplicitGC -XX:TargetSurvivorRatio=90 -XX:G1NewSizePercent=50 -XX:G1MaxNewSizePercent=80 -XX:G1MixedGCLiveThresholdPercent=35 -XX:+AlwaysPreTouch -XX:+ParallelRefProcEnabled -Dusing.aikars.flags=mcflags.emc.gs -jar server.jar --nojline
```

**Comment by electronicboy** [comment]
How are you running the server? are you using some form of wrapper or terminal multiplexor?

**Comment by brianewell** [comment]
@electronicboy I've tried it with and without tmux: same behavior both ways.

**Comment by zachbr** [comment]
Edit: This will work fine from the **host/root** zone but it will not work properly in the **guest** zone.
This issue is confirmed.

---

I can't seem to reproduce this unfortunately. I grabbed what I think is the correct download from the SmartOS site but it appears to be a nightly build. Is there a repository of stable builds somewhere I should be testing against instead?

I'm afraid I don't have much background in Solaris/SunOS. Any additional info you can think of would be appreciated.
 
`SunOS smartos-vm 5.11 joyent_20181108T015750Z i86pc i386 i86pc`
`java version "1.8.0_191"
Java(TM) SE Runtime Environment (build 1.8.0_191-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.191-b12, mixed mode)`
`Paper version git-Paper-451 (MC: 1.13.2) (Implementing API version 1.13.2-R0.1-SNAPSHOT)`

![](https://i.imgur.com/YRbGyrn.png)

**Comment by brianewell** [comment]
@zachbr wow, thanks for diving into this so quickly.

SmartOS has a bi-weekly release schedule, but I'm back on the 20181011 release; I'll spin up a VM with an updated image and see if that helps at all.  Also, SmartOS is a hypervisor, and at first glance it looks like you're running paper in the global zone instead of a guest zone.  I'll try to replicate on the global zone and see what happens.

**Comment by zachbr** [comment]
That is correct, I was in the root zone. I will try and test in a guest zone later as well. And thanks for the additional info.

**Comment by brianewell** [comment]
Quick steps to get a guest zone running:

1. Download a guest image
`imgadm import c6a275e4-c730-11e8-8c5f-9b24fe560a8f`

2. Instance the guest image
```
vmadm create <<EOF
{
        "image_uuid": "c6a275e4-c730-11e8-8c5f-9b24fe560a8f",
        "brand": "joyent",
        "hostname": "minecraft",
        "alias": "minecraft",
        "max_physical_memory": 4096,
        "resolvers": [ "1.1.1.1" ],
        "nics": [{
                "nic_tag": "admin",
                "ip": "dhcp",
                "primary": true
        }],
        "customer_metadata": {
                "root_authorized_keys": "your ssh key here",
                "user-script": "/usr/sbin/mdata-get root_authorized_keys > /root/.ssh/authorized_keys ; /usr/sbin/mdata-get root_authorized_keys > /home/admin/.ssh/authorized_keys"
        }
}
EOF
```

3. zlogin to the returned uuid.  You can also ssh directly once you know the IP address.

edit: added max_physical_memory to the instance json, the default of 256M is entirely too small.

**Comment by zachbr** [comment]
This is an issue with the terminal input library we use, I've submitted a ticket to them to see how/if they'd like to deal with this issue. We'll see how we should proceed based on how they respond.

https://github.com/jline/jline3/issues/330

Thanks for letting us know. In the meantime, you should continue to use the `-nojline` option.

**Comment by brianewell** [comment]
I've tested with the following JVMs within a guest zone and had exceptions raised:
* Java HotSpot(TM) 64-Bit Server VM 25.191-b12
* OpenJDK 64-Bit Server VM 25.181-b00

I've also tested with HotSpot in the global zone with no issues whatsoever, so I'm going to assume this is actually a issue between the environment and the JRE.  Thanks for your attention on this matter.

**Comment by zachbr** [comment]
As an update to this ticket, the issue has been fixed in the terminal input library we use (JLine) and they are going to try and get a release out before the end of January.

Once that release is out I'll see about poking Minecrell (no ping) into updating TerminalConsoleAppender and then we can update Paper.

Thanks for your patience.

**Comment by zachbr** [comment]
This is fixed in the 1.14.x builds. Probably not going to get backported to 1.13.2 due to some issues with Windows in that version.
